Fix failing multiprocessing UT

shreyamalviya opened the pull request:
Fixes #1 
Resolves https://github.com/guardicore/monkey/issues/2684

The culprit of the mulitprocessing plugin tests failing is actually the `test_module_not_found_incorrect_plugin_name()` test. If you remove it and run the test suite, everything passes. 

A detailed explanation can be found in the commit message of the first commit of this PR (https://github.com/guardicode/serpentarium/commit/f67b3fa6e3e29f2cc8a8c0bf73d7d711654c9c42).

shreyamalviya committed:
Remove comment about failing test in test_plugin_loader.py

mssalvatore reviewed (commented):
Great job on the root cause analysis. Is there a way to trap the exception and make sure `_clean_system_modules()` runs? This might be a more robust solution.

shreyamalviya committed:
Catch and reraise exceptions in `PluginWrapper.run()` so context cleanup finishes properly

The try/except block catches the raised exception. The context manager
is allowed to exit so that cleanup happens as expected. Then, the caught
exception is raised again.

mssalvatore committed:
Add test to ensure sys.modules is restored even if exception raised

mssalvatore committed:
Simplify exception handling logic in PluginWrapper.run()

mssalvatore committed:
Extract method PluginWrapper._run_in_isolated_context()

mssalvatore committed:
Add a comment about skipping context switching in PluginWrapper.run()

mssalvatore committed:
Add a changelog entry for #1
